packages/web-wallet/src/hooks/useTokens.ts

```diff
@@ -0,0 +1,82 @@
+import { useMemo } from 'react';
+import { useWallet } from '../contexts/WalletContext';
+import type { TokenInfo, TokenFilter } from '../types/token';
+import { TOKEN_IDS, HTR_TOKEN_INFO } from '../constants';
+
+/**
+ * Custom hook for token operations and filtering
+ */
+export function useTokens(selectedTokenFilter: TokenFilter = 'all') {
+  const {
+    registeredTokens,
+    registerToken,
+    unregisterToken,
+    refreshTokenBalances,
+    getTokenBalance,
+    network,
+    balances,
+  } = useWallet();
+
+  // Create HTR token info from balances
+  const htrToken: TokenInfo = useMemo(() => {
+    const htrBalance = balances.length > 0 ? balances[0] : { available: 0n, locked: 0n };
+    return {
+      uid: TOKEN_IDS.HTR,
+      name: HTR_TOKEN_INFO.name,
+      symbol: HTR_TOKEN_INFO.symbol,
+      balance: {
+        available: htrBalance.available,
+        locked: htrBalance.locked,
+      },
+      isNFT: false,
+      registeredAt: 0, // HTR is always available
+    };
```

Comment:
An idea would be to have the token data and metadata separate.
This would mean that token data list would be more stable and would change less often.
Metadata may change as much as needed but will not require updates on other stuff.